src/main/java/com/example/bistro/bistroapp/console/BistroM.java

```diff
@@ -0,0 +1,338 @@
+                    String lastName = scanner.next();
+
+                    Customer newCustomer = new Customer();
+                    newCustomer.setFirstName(firstName);
+                    newCustomer.setLastName(lastName);
+
+                    Customer addedCustomer = customerService.addCustomer(newCustomer);
+                    System.out.println("Added customer with ID: " + addedCustomer.getId());
+                    break;
+                case 7:
+                    List<Customer> allCustomers = customerService.getAllCustomers();
+                    System.out.println("All Customers:");
+                    for (Customer customer : allCustomers) {
+                        System.out.println(customer.getId() + ": " + customer.getFirstName() + " " + customer.getLastName());
+                    }
+                    break;
+                case 8:
+                    System.out.println("Enter the customer ID:");
+                    Long customerIdToRemove = scanner.nextLong();
+
+                    try {
+                        customerService.removeCustomer(customerIdToRemove);
+                        System.out.println("Removed customer with ID " + customerIdToRemove);
+                    } catch (Exception e) {
+                        System.out.println("Error removing customer: " + e.getMessage());
+                    }
+                    break;
+                case 9:
+    int topCount = 3;
+    List<Product> topProducts = productService.getTopMostWantedProducts(topCount);
+    System.out.println("Top " + topCount + " Most Popular Products:");
+    for (Product product : topProducts) {
+        System.out.println(product.getName());
+    }
+    break;
+
+                case 10:
+                    introducereComanda();
+                    break;
+                case 11:
+                    afiseazaToateComenzile();
+                    break;
+                case 12:
+                    Set<Ingredient> addedIngredients = introducereIngrediente();
+                    for (Ingredient ingredient : addedIngredients) {
+                        ingredientService.addIngredient(ingredient);
+                    }
+                    System.out.println("Ingredients added successfully.");
+                    break;
+                case 13:
+                    List<Ingredient> allIngredients = ingredientService.getAllIngredients();
+                    System.out.println("All Ingredients:");
+                    for (Ingredient ingredient : allIngredients) {
+                        System.out.println(ingredient.getId() + ": " + ingredient.getName());
+                    }
+                    break;
+                case 0:
+                    running = false;
+                    break;
+                default:
+                    System.out.println("Invalid choice. Please try again.");
+
+            }
+        }
+        scanner.close();
+    }
+    @Transactional
```

Comment:
Why do you need this annotation here? You should use it only in service layer if it's necessary.